pkg/render/node.go

```diff
@@ -1431,8 +1445,9 @@ func (c *nodeComponent) nodeEnvVars() []corev1.EnvVar {
 		{Name: "NO_DEFAULT_POOLS", Value: "true"},
 	}
 
-	// Only append goldmane variables if goldmane is running.
-	if c.cfg.GoldmaneRunning {
+	// Only append goldmane variables if goldmane is running and we have a valid IP address,
+	// as we rely on an explicitly configured host alias to resolve the goldmane service.
+	if c.cfg.GoldmaneRunning && c.cfg.GoldmaneIP != "" {
```

Comment:
I don't think this is a change so not a comment, but it would be preferable if we could somehow not restart Felix when Goldmane starts.  It's early in the day (so to speak), so not a huge deal, but ideally we would restart calico-node as little as possible.  It's not completely graceful (some flow logging data is lost).